src/Diagrams/Backend/SVG.hs

```diff
@@ -339,9 +346,54 @@ attributedRender svg = do
   clippedSvg   <- renderSvgWithClipping preT svg sty
   lineGradDefs <- lineTextureDefs sty
   fillGradDefs <- fillTextureDefs sty
+  let (transOriginAttr, transElement) = case getAttr sty of
+        Nothing -> ([], mempty)
+        Just (TransformAnimationAttribute animations t) ->
+          ( pure . makeAttribute "transform-origin" . (\(P (V2 x y)) -> showNum x <> " " <> showNum y) $ transform t 1
+          , flip foldMap animations $ \(TransformAnimation dur rep animation) -> animateTransform_ $
+              [ AttributeName_ <<- "transform"
+              , Additive_ <<- "sum"
+              , Dur_ <<- T.show dur <> "s"
+              , RepeatCount_ <<- maybe "indefinite" T.show rep
+              ] <> case animation of
+                ScaleAnimation values ->
+                  [ Type_ <<- "scale"
+                  , Values_ <<- T.intercalate ";" (map (\(V2 x y) -> showNum x <> "," <> showNum y) values)
+                  ]
+                TranslateAnimation values ->
+                  [ Type_ <<- "translate"
+                  , Values_ <<- T.intercalate ";" (map (\(V2 x y) -> showNum x <> "," <> showNum y) values)
+                  ]
+            )
   return $ do
     let gDefs = mappend fillGradDefs lineGradDefs
-    gDefs `mappend` g_ (R.renderStyles idFill idLine sty) clippedSvg
+    gDefs `mappend` g_ (transOriginAttr <> R.renderStyles idFill idLine sty) (transElement <> clippedSvg)
+  where
+    showNum = T.pack . showFixed @E3 True . realToFrac
```

Comment:
I've gone with three decimal places based on looking at existing numbers in the SVG output. Presumably there's some code for those which I should be reusing here?

Reply:
I don't remember!